### PR TITLE
Add return code to source_foreign functions

### LINKED ISSUE #817: source-bash return code will always be 0
Even if the file doesn't exist.

### PR DESCRIPTION
**Pull request by melund**
This fixes #817 by adding a return code to source_foreign

### CODE REVIEW

**Comment by melund** [comment]
I also added a non-zero return value to the xexec alias if the commands fails

**Comment by gforsyth** [comment]
Hey @melund -- I'm still getting a `0` returncode when I `source-bash` a nonexistent file.

```console
(xonshtest2)gil@theo~/tmp$ ls
foo
(xonshtest2)gil@theo~/tmp$ source-bash bar
(xonshtest2)gil@theo~/tmp$ __xonsh_history__.rtns[-1]
0
```

File: xonsh/aliases.py [hunk 1]
```diff
@@ -282,7 +284,7 @@ def xexec(args, stdin=None):
         except FileNotFoundError as e:
             return 'xonsh: ' + e.args[1] + ': ' + args[0] + '\n'
```
Comment (scopatz) [inline xonsh/aliases.py]: This line could also use a non-zero return code while we are at it :)

**Comment by scopatz** [comment]
@melund, First off, thanks a million for putting this in!

@gforsyth @melund - my guess is that this is because we are not running `set -e` at the top of the Bash script that get created in `foreign_shell_data()`.  Thus, the non-existent file is name is being passed to a script that looks like:

```bash
doesntexist  # fails
echo __XONSH_ENV_BEG__  #runs anyway
env
echo __XONSH_ENV_END__
...
```

Rather than 

```bash
set -e
doesntexist  # fails and exits
echo __XONSH_ENV_BEG__
env
echo __XONSH_ENV_END__
...
```

Thus the function actually does succeed.  We should probably implement this fail-on-error semantic.  Not sure that this PR is the right place for it.  Happy to merge this as-is, even though it doesn't solve the original problem yet.

File: xonsh/aliases.py (hunk 1 above)
Comment (melund) [inline xonsh/aliases.py]: oh yes... Maybe we shouldn't add non-zero return codes to this function. It is the equivalent to python exec so it should only use Exceptions. What is you thought on that?

Comment (melund) [inline xonsh/aliases.py]: Or should it do both?

Comment (melund) [inline xonsh/aliases.py]: Well,  I guess it should just return a non-zero return code. I will add that.

**Comment by melund** [comment]
Sorry for beeing a too quick at pushing this. This actually doesn't work on Win either. The bat file need a post command with `if errorlevel 1 exit 1`.. I have added that now.

I think someone else should make a PR with the `set -e` stuff. I don't have the linux box to test it on.

**Comment by melund** [comment]
@scopatz .. I think this is ready for merge. Maybe you can add and test the `set -e`stuff.

**Comment by scopatz** [comment]
Thanks a ton @melund!  I'll try adding the `set -e` stuff.

**Pull request merged**